### PR TITLE
validate round number on returned beacons

### PR DESCRIPTION
**Pull request by lourkeur**
`make test` passing locally.

### CODE REVIEW

File: client/verify.go [hunk 1]
```diff
@@ -57,6 +57,9 @@ func (v *verifyingClient) Get(ctx context.Context, round uint64) (Result, error)
 	if err := v.verify(ctx, info, rd); err != nil {
 		return nil, err
 	}
+	if round != 0 && rd.Round() != round {
+		return nil, fmt.Errorf("round misatch: %d != %d", rd.Round(), round)
```
Comment (CluEleSsUK) [inline client/verify.go:61]:
```suggestion
		return nil, fmt.Errorf("round mismatch: %d != %d", rd.Round(), round)
```

File: client/verify_test.go [hunk 2]
```diff
@@ -51,3 +51,9 @@ func VerifyFuncTest(t *testing.T, clients, upTo int) {
 		t.Fatal("expected to get result.", results[upTo].Round(), res.Round(), fmt.Sprintf("%v", c))
 	}
 }
+
+func TestGetWithRoundMismatch(t *testing.T) {
+	c, _ := mockClientWithVerifiableResults(t, 3, false)
+	_, err := c.Get(context.Background(), 2)
+	require.ErrorContains(t, err, "round misatch: 1 != 2")
```
Comment (CluEleSsUK) [inline client/verify_test.go:58]:
```suggestion
	require.ErrorContains(t, err, "round mismatch: 1 != 2")
```

**Review by CluEleSsUK — COMMENTED** [review]
Thanks for your contribution! Have suggested some small spelling changes, but otherwise good to go.
We're releasing drand v2.0.0 shortly which includes extraction of the client code into its own repo, so we'll need to backport the change there

**Comment by lourkeur** [comment]
seems like there is some kind of indeterminism. That explains the CI status. Investigating

```
go test ./client/ -test.run TestGetWithRoundMismatch -count=8 -v
[...]
2023-12-17T12:07:49.528+0100	INFO	client/optimizing.go:194		{"optimizing_client": "endpoint down when speed tested", "client": "MockInfo.(+verifier)", "err": "not supported (mock info client get)"}
    verify_test.go:58: 
        	Error Trace:	/home/louis/dev/drand/client/verify_test.go:58
        	Error:      	An error is expected but got nil.
        	Test:       	TestGetWithRoundMismatch
--- FAIL: TestGetWithRoundMismatch (0.01s)
```

**Comment by CluEleSsUK** [comment]
Fear not, some of our tests on master are a bit flaky -.-‘ drand v2.0.0 will solve all our problems (tm)

**Comment by lourkeur** [comment]
The speed test from the optimizing client is interfering. This probably shouldn't use `client.Wrap`...

**Review by CluEleSsUK — APPROVED** [review]
Thanks again for your contribution!

**Pull request merged**